### DISCUSSION TITLE
What is "Steam API key integration" worth or used for?

### DISCUSSION THREAD

**Discussion by dmanas0** [akiver/cs-demo-manager#667] [Q&A] [answered]
What is "Steam API key integration" worth or used for?

**Comment by akiver** [comment, marked as answer]
Hi, I added a documentation [here](https://cs-demo-manager.com/docs/guides/settings#third-party-api-keys).
Does it answer your question? My recommendation is to generate a key.

**Reply by dmanas0** [reply]
Sure but what's the difference between not having it? I see it retrieves the Valve matches the same way. Does it retrieve some additional information with the API Key or something?

**Reply by akiver** [reply]
If you don't use a custom API key you may have a notification saying "Steam API returned a 429 status code" because the key bundled in the application reached the rate limit.
Valve matches are not retrieved with the Steam web API but with [this tool](https://github.com/akiver/boiler-writter) - the key is used to retrieve VAC ban stats and players' avatars.

**Reply by dmanas0** [reply]
Now I understand. Thanks.
